### PR TITLE
Add docs and tests for the WebRTC components

### PR DESCRIPTION
**Pull request by ipavlidakis**
### 📝 Summary

This revision adds documentation and test for all remaining WebRTC components.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamVideo/WebRTC/v2/StateMachine/Stages/WebRTCCoordinator+Joined.swift [hunk 1]
```diff
@@ -385,17 +385,20 @@ extension WebRTCCoordinator.StateMachine.Stage {
...
+            if await stateAdapter.statsReporter?.sessionID != sessionId {
+                let statsReporter = WebRTCStatsReporter(
+                    sessionID: await stateAdapter.sessionID
+                )
+                await stateAdapter.set(statsReporter)
+            }
 
-            await stateAdapter.set(statsReporter)
+            let statsReporter = await stateAdapter.statsReporter
+            statsReporter?.interval = await stateAdapter.statsReporter?.interval ?? 0
```
Comment (martinmitrevski) [inline Sources/StreamVideo/WebRTC/v2/StateMachine/Stages/WebRTCCoordinator+Joined.swift:398]: if it's a zero, we don't report, right?

File: Sources/StreamVideo/WebRTC/v2/WebRTCStateAdapter.swift [hunk 2]
```diff
@@ -6,13 +6,20 @@ import Combine
 import Foundation
 import StreamWebRTC
 
+/// An actor class that handles WebRTC state management and media tracks for a
+/// video call. This class manages the connection setup, track handling, and
+/// participants, including their media settings, capabilities, and track
+/// updates.
+
```
Comment (martinmitrevski) [inline Sources/StreamVideo/WebRTC/v2/WebRTCStateAdapter.swift:13]: extra space here

File: StreamVideoTests/WebRTC/v2/WebRTCAuthenticator_Tests.swift [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+            }
+
+            try await group.waitForAll()
+        }
+    }
+
+    // MARK: - waitForConnect
+
+    func test_waitForConnect_shouldThrowErrorIfTimeout() async throws {
+        do {
```
Comment (martinmitrevski) [inline StreamVideoTests/WebRTC/v2/WebRTCAuthenticator_Tests.swift:219]: wasn't there a way to assert throwing in XCTest? We could skip the do/catch and fail.

File: StreamVideoTests/WebRTC/v2/WebRTCCoorindator_Tests.swift [hunk 4]
```diff
@@ -0,0 +1,565 @@
...
+            false
+        )
+    }
+
+    // MARK: - changeVideoState
+
+    func test_changeVideoState_shouldUpdateVideoState() async throws {
+        /// This can't be test due to the preprocessor macro we are using
+        /// in the CallSettings init.
+//        await subject.changeVideoState(isEnabled: false)
```
Comment (martinmitrevski) [inline StreamVideoTests/WebRTC/v2/WebRTCCoorindator_Tests.swift:165]: Ah, I remember that :) Still would prefer to remove the test over having commented out code.

File: StreamVideoTests/WebRTC/v2/WebRTCStateAdapter_Tests.swift [hunk 5]
```diff
@@ -0,0 +1,663 @@
...
+            name: .unique,
+            filter: { _ in fatalError() }
+        )
+        await subject.set(sfuAdapter: sfuStack.adapter)
+        await subject.set(videoFilter)
+        await subject.set(ownCapabilities)
+        await subject.set(callSettings)
+        await subject.set(.unique)
+        await subject.set(token: .unique)
+        await subject.set(12)
```
Comment (martinmitrevski) [inline StreamVideoTests/WebRTC/v2/WebRTCStateAdapter_Tests.swift:314]: why don't we use named params? It's a bit tricky to figure out what this sets.

File: StreamVideoTests/WebRTC/v2/WebRTCStateAdapter_Tests.swift [hunk 6]
```diff
@@ -0,0 +1,663 @@
...
+        await assertEqualAsync(await subject.anonymousCount, 0)
+        await assertEqualAsync(await subject.participantPins, [])
+    }
+
+    // MARK: - cleanUpForReconnection
+
+    func test_cleanUpForReconnection_shouldResetPropertiesForReconnection() async throws {
+        let sfuStack = MockSFUStack()
+        sfuStack.setConnectionState(to: .connected(healthCheckInfo: .init()))
+        let ownCapabilities = Set([OwnCapability.blockUsers, .changeMaxDuration])
```
Comment (martinmitrevski) [inline StreamVideoTests/WebRTC/v2/WebRTCStateAdapter_Tests.swift:346]: can we move this chunk into a helper method, since it's repeatable?

**Review by martinmitrevski — APPROVED** [review]
great work ✅ Added few small things, but feel free to merge.

File: Sources/StreamVideo/WebRTC/v2/StateMachine/Stages/WebRTCCoordinator+Joined.swift (hunk 1 above)
Comment (ipavlidakis) [reply Sources/StreamVideo/WebRTC/v2/StateMachine/Stages/WebRTCCoordinator+Joined.swift:398]: That's right. This is the method that runs whenever the interval property is changing on the `WebRTCStatsReporter`
```swift
private func scheduleCollection(with interval: TimeInterval) {
        guard interval > 0 else {
            log.warning("Collection interval should be greater than 0.")
            collectionCancellable?.cancel()
            return
        }

        collectionCancellable?.cancel()
        collectionCancellable = Foundation
            .Timer
            .publish(every: interval, on: .main, in: .default)
            .autoconnect()
            .sink { [weak self] _ in self?.collectStats() }

        log.debug("Stats collection is now scheduled with interval:\(interval).")
    }
```

File: StreamVideoTests/WebRTC/v2/WebRTCAuthenticator_Tests.swift (hunk 3 above)
Comment (ipavlidakis) [reply StreamVideoTests/WebRTC/v2/WebRTCAuthenticator_Tests.swift:219]: Good point

**Pull request merged**